securedrop_client/api_jobs/base.py

```diff
@@ -91,3 +91,23 @@ def call_api(self, api_client: API, session: Session) -> Any:
         failure.
         '''
         raise NotImplementedError
+
+
+class SingleObjectApiJob(ApiJob):
+    def __init__(self, uuid: str, remaining_attempts: int = DEFAULT_NUM_ATTEMPTS) -> None:
+        super().__init__(remaining_attempts)
+        '''
+        UUID of the item (source, reply, submission, etc.) that this item
+        corresponds to. We track this to prevent the addition of duplicate jobs.
+        '''
+        self.uuid = uuid
```

Comment:
i see, since we have an api job that performs a task on many objects, i.e. `MetadataSyncJob`, which gets all remote source data, we can't just extend `ApiJob` to have a uuid. the rest of our jobs are "single object" jobs, so they'll each have a uuid associated with them. this does get me thinking about batch jobs and checking for duplicate lists of uuids to prevent duplicate batch jobs, but we don't have to worry about this yet.

also, since this change prevents any SingleObjectApiJob from entering the queue if we're already performing an operation on that object, i.e. if a job with that object's uuid is already in the queue, we have to make sure we only ever derive from this class if we are sure there are no other type of operations that could happen on the same object simultaneously.